query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala

```diff
@@ -336,8 +336,14 @@ class QueryResource(secondary: Secondary,
               case PipeQuery(l, r) =>
                 val (nl, rollupLeft) = possiblyRewriteOneAnalysisInQuery(schema, l, Some(ruMap))
                 val (nr, rollupJoin) = possiblyRewriteJoin(r)
-                val rewritten = (PipeQuery(nl, nr), rollupLeft ++ rollupJoin)
-                if (ruMapOpt.isEmpty && ruMap.nonEmpty && rollupLeft.isEmpty) {
+                val (nr2, rollupRight) = possiblyRewriteOneAnalysisInQuery(schema, nr, Some(ruMap))
```

Comment:
At this point we have tried to "rewrite joins" of the right pipe side.
Now we additionally need to pass this through a round of "rewrite to rollup".